Framework/Algorithms/src/CalculateEfficiency.cpp

```diff
@@ -273,5 +264,37 @@ void CalculateEfficiency::normalizeDetectors(MatrixWorkspace_sptr rebinnedWS,
   }
 }
 
+/**
+ * Fully masks one component named componentName
+ */
+void CalculateEfficiency::maskComponent(MatrixWorkspace &ws,
+                                        const std::string &componentName) {
+  auto instrument = ws.getInstrument();
+  try {
+    boost::shared_ptr<const Geometry::ICompAssembly> component =
+        boost::dynamic_pointer_cast<const Geometry::ICompAssembly>(
+            instrument->getComponentByName(componentName));
```

Comment:
Would it be safer to check here that `component` is fine (as they do below for `detector`)?